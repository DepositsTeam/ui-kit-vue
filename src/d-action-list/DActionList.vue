<template>
  <d-box class="ui-action-list">
    <d-box class="ui-action-list__header">
      <d-box class="ui-action-list__header__left">
        <d-text font-face="circularSTD">{{ label }}</d-text>
      </d-box>
      <d-box class="ui-action-list__header__left">
        <slot name="call-to-action">
          <d-button
            colorScheme="primary"
            :left-icon="AddIcon"
            :size="size"
            @click="emit('call-to-action-clicked')"
          >
            {{ callToActionText }}
          </d-button>
        </slot>
      </d-box>
    </d-box>
    <d-box class="ui-action-list__body__wrapper">
      <slot name="body">
        <d-box
          v-for="(application, index) in list"
          :key="`application__${index}`"
        >
          <d-action-list-item
            :logo="application.logo"
            :title="application.title"
            :subtitle="application.subtitle"
            :description="application.description"
            :button-size="application.buttonSize"
            :button-color-scheme="application.buttonColorScheme"
            :button-smart-color="application.buttonSmartColor"
            buttonText="Revoke Access"
            @clicked="listActionClicked(application)"
          />
        </d-box>
      </slot>
    </d-box>
  </d-box>
</template>

<script setup>
import { DBox, DText, DButton, AddIcon, DActionListItem } from "@/main";

const emit = defineEmits(["call-to-action-clicked", "list-action-clicked"]);
defineProps({
  label: {
    type: String,
    required: true,
  },
  list: {
    type: Array,
    required: true,
  },
  size: {
    type: String,
    default: "large",
  },
  callToActionText: {
    type: String,
    default: "New Connection",
  },
});

const listActionClicked = (application) => {
  emit("list-action-clicked", application);
};
</script>

<style scoped lang="scss">
.ui-action-list {
  .ui-action-list__header {
    display: flex;
    padding: 16px 24px;
    justify-content: space-between;
    border-radius: 8px 8px 0 0;
    border: 0.5px solid #e1e7ec;
    background: #fff;
    align-self: stretch;

    .ui-action-list__header__left {
      color: #8c97a7;
      font-size: 14px;
      font-style: normal;
      font-weight: 450;
      line-height: normal;
    }
  }

  .ui-action-list__body__wrapper {
    .ui-action-list__body {
      display: flex;
      padding: 16px 24px;
      justify-content: space-between;
      align-items: center;
      align-self: stretch;
      border: 0.5px solid #e2e8f0;
      background: #fff;

      .ui-action-list__logo {
        height: 32px;
        border-radius: 50%;
      }

      &:last-child {
        border-radius: 0 0 8px 8px;
      }

      .ui-action-list__header__left {
        display: flex;
        align-items: center;
        gap: 8px;

        .d-action-card__logo {
          width: 32px;
          height: 32px;
          border-radius: 50%;
        }
      }

      .ui-action-list__header__left {
        display: flex;
        align-items: center;
        gap: 16px;

        .ui-action-list__subtitle-text {
          color: #8c97a7;
          font-size: 14px;
          font-style: normal;
          font-weight: 450;
          line-height: 24px; /* 171.429% */
        }
      }
    }
  }
}
</style>
