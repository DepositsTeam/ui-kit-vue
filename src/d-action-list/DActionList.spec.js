import { mount } from "@vue/test-utils";
import { expect, test } from "vitest";
import DActionList from "./DActionList.vue";

test("renders the component", async () => {
  expect(DActionList).to.not.be.undefined;

  const wrapper = mount(DActionList, {
    props: {
      label: "Application",
      list: [
        {
          id: 1,
          title: "example@gmail.com",
          description: "Duration: 1hr | Expires: July 27, 2021 12:45 PM",
          logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
          buttonText: "Revoke access",
          subtitle: "Granted, July 27, 2021 12:45 PM",
          buttonSize: "large",
        },
        {
          id: 2,
          title: "example@gmail.com",
          description: "Duration: 1hr | Expires: July 27, 2021 12:45 PM",
          logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
          buttonText: "Revoke access",
          subtitle: "Granted, July 27, 2021 12:45 PM",
          buttonSize: "large",
        },
      ],
    },
  });

  expect(wrapper.get(".ui-action-list__header__left").text()).to.be.equal(
    "Application"
  );

  expect(wrapper.findAll('data-test="action-list-item"]')).to.have.lengthOf(2);
});

test("updates call to action", async () => {
  const wrapper = mount(DActionList, {
    props: {
      label: "Application",
      callToActionText: "Revoke all access",
      list: [
        {
          id: 1,
          title: "example@gmail.com",
          description: "Duration: 1hr | Expires: July 27, 2021 12:45 PM",
          logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
          buttonText: "Revoke access",
          subtitle: "Granted, July 27, 2021 12:45 PM",
          buttonSize: "large",
        },
        {
          id: 2,
          title: "example@gmail.com",
          description: "Duration: 1hr | Expires: July 27, 2021 12:45 PM",
          logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
          buttonText: "Revoke access",
          subtitle: "Granted, July 27, 2021 12:45 PM",
          buttonSize: "large",
        },
      ],
    },
  });

  expect(
    wrapper.find('[data-test="call-to-action-button"]').text()
  ).to.be.equal("Revoke all access");
});

test("renders the component without list prop", async () => {
  const wrapper = mount(DActionList, {
    props: {
      label: "Application",
    },
  });

  expect(wrapper.get(".ui-action-list__header__left").text()).to.be.equal(
    "Application"
  );

  expect(wrapper.find('[data-test="action-list-item"]').exists()).to.be.false;
});

test("matches snapshot", async () => {
  const wrapper = mount(DActionList);
  expect(wrapper.element).toMatchSnapshot();
});

test("slots work as expected", async () => {
  const wrapper = mount(DActionList, {
    slots: {
      default: "I am the expanded content",
      callToAction: "Button content",
    },
  });

  expect(wrapper.get(".ui-action-list__body__wrapper").text()).to.be.equal(
    "I am the expanded content"
  );

  expect(wrapper.get(".ui-action-list__header__right").text()).to.be.equal(
    "Button content"
  );
});

test("hides call to action", async () => {
  const wrapper = mount(DActionList, {
    props: {
      label: "Application",
      callToActionText: "Revoke all access",
      hideCallToAction: true,
      list: [
        {
          id: 1,
          title: "example@gmail.com",
          description: "Duration: 1hr | Expires: July 27, 2021 12:45 PM",
          logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
          buttonText: "Revoke access",
          subtitle: "Granted, July 27, 2021 12:45 PM",
          buttonSize: "large",
        },
        {
          id: 2,
          title: "example@gmail.com",
          description: "Duration: 1hr | Expires: July 27, 2021 12:45 PM",
          logo: "https://img.freepik.com/free-vector/golden-bird-logo-design_1195-336.jpg",
          buttonText: "Revoke access",
          subtitle: "Granted, July 27, 2021 12:45 PM",
          buttonSize: "large",
        },
      ],
    },
  });

  expect(wrapper.find('[data-test="call-to-action-button"]').exists()).to.be
    .false;
});
