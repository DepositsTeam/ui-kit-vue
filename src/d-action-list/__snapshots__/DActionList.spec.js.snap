// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  class="circularSTD deposits-ui-box ui-action-list"
  data-ui-kit-auto_generated-cltvdguj3s=""
  data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
  data-v-e5c24b7e=""
>
  
  <div
    class="circularSTD deposits-ui-box ui-action-list__header"
    data-ui-kit-auto_generated-cltvdguj3s=""
    data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
    data-v-e5c24b7e=""
  >
    
    <div
      class="circularSTD deposits-ui-box ui-action-list__header__left"
      data-ui-kit-auto_generated-cltvdguj3s=""
      data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
      data-v-e5c24b7e=""
    >
      
      <p
        center="false"
        class="circularSTD deposits-ui-box ui-text"
        data-ui-kit-auto_generated-rkysuan9l1=""
        data-ui-kit-theming-rules-auto_generated-rkysuan9l1=""
        data-v-e5c24b7e=""
        equallineheight="false"
        mb0="false"
        ml0="false"
        mr0="false"
        mt0="false"
        mx0="false"
        my0="false"
        uppercase="false"
      >
        
        
        
        
        
      </p>
      
    </div>
    <div
      class="circularSTD deposits-ui-box ui-action-list__header__right"
      data-ui-kit-auto_generated-cltvdguj3s=""
      data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
      data-v-e5c24b7e=""
    >
      
      
      <button
        class="heroNew deposits-ui-box ui-button semantic__primary size__large"
        data-test="call-to-action-button"
        data-ui-kit-auto_generated-c1xbggyioe=""
        data-ui-kit-theming-rules-auto_generated-c1xbggyioe=""
        data-v-5100a3c9=""
        data-v-e5c24b7e=""
      >
        
        <div
          class="circularSTD deposits-ui-box"
          data-ui-kit-auto_generated-wh6g3feqwh=""
          data-ui-kit-theming-rules-auto_generated-wh6g3feqwh=""
          data-v-5100a3c9=""
        >
          
          <svg
            class="deposits-ui-box ui-button__left-icon"
            data-ui-kit-auto_generated-7LxGHMe8z3=""
            data-ui-kit-theming-rules-auto_generated-7LxGHMe8z3=""
            data-v-5100a3c9=""
            fill="none"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            
            <path
              d="M12 5.75V18.25"
              stroke="currentcolor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M18.25 12H5.75"
              stroke="currentcolor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            
          </svg>
          <div
            class="circularSTD deposits-ui-box ui-button__button-text"
            data-ui-kit-auto_generated-cltvdguj3s=""
            data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
            data-v-5100a3c9=""
          >
            
            <div
              class="circularSTD deposits-ui-box"
              data-ui-kit-auto_generated-cltvdguj3s=""
              data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
              data-v-5100a3c9=""
            >
              
              
              New Connection
              
              
            </div>
            
          </div>
          <!--v-if-->
          <!--v-if-->
          
        </div>
        
      </button>
      
      
    </div>
    
  </div>
  <div
    class="circularSTD deposits-ui-box ui-action-list__body__wrapper"
    data-ui-kit-auto_generated-cltvdguj3s=""
    data-ui-kit-theming-rules-auto_generated-cltvdguj3s=""
    data-v-e5c24b7e=""
  >
    
    
  </div>
  
</div>
`;
